docs/glossary.asciidoc

```diff
@@ -0,0 +1,52 @@
+[id="{p}-glossary"]
+== Glossary
+
+This glossary supplements the link:https://kubernetes.io/docs/reference/glossary/[Kubernetes glossary] and covers terms used in the Elastic Cloud on Kubernetes (ECK) documentation.
+
+[[CA]]CA::
+  Certificate Authority. An entity that issues digital certificates to verify identities over a network.
+
+[[Cluster]]Cluster::
+  Can refer to either an link:https://www.elastic.co/guide/en/elasticsearch/reference/current/add-elasticsearch-nodes.html[Elasticsearch cluster] or a Kubernetes cluster depending on the context.
+
+[[CRD]]CRD::
+  link:https://kubernetes.io/docs/reference/glossary/?fundamental=true#term-CustomResourceDefinition[Custom Resource Definition]. ECK extends the Kubernetes API with CRDs to allow users to deploy and manage Elasticsearch, Kibana and APM server resources just as they would do with built-in Kubernetes resources.
+
+[[ECK]]ECK::
+  Elastic Cloud on Kubernetes. Kubernetes operator to orchestrate Elasticsearch, Kibana and APM Server deployments on Kubernetes.
+
+[[EKS]]EKS::
+  Elastic Kubernetes Service. Managed Kubernetes service provided by Amazon Web Services (AWS).
+
+[[GCS]]GCS::
+  Google Cloud Storage. Block storage service provided by Google Cloud Platform (GCP).
+
+[[GKE]]GKE::
+  Google Kubernetes Engine. Managed Kubernetes service provided by Google Cloud Platform (GCP).
+
+[[k8s]]k8s::
```

Comment:
I think it needs to be `K8s` with a capitalized `K`, according to https://github.com/cncf/foundation/blob/master/style-guide.md#2-kubernetes